test/e2e/tests/load_balancing.go

```diff
@@ -358,3 +361,159 @@ var ConsistentHashCookieLoadBalancingTest = suite.ConformanceTest{
+				cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+				require.NoError(t, err, "failed to get expected response")
+				require.NoError(t, http.CompareRequest(t, &req, cReq, cResp, expectedResponse), "failed to compare request and response")
+
+				actualPodName := cReq.Pod
+				require.Equal(t, expectPodName, actualPodName, "request %d: expected pod %s but got %s", i+1, expectPodName, actualPodName)
+			}
+
+			t.Logf("All %d requests with valid override host %s routed to expected pod: %s", sendRequests, overrideHost, expectPodName)
+		})
+		t.Run("header-based endpoint override with invalid pod IP should fallback to load balancer policy", func(t *testing.T) {
+			// Use an invalid pod IP that's not in the service endpoints
+			invalidOverrideHost := "192.168.99.99:8080"
+
+			t.Logf("Testing endpoint override with invalid pod IP: %s (should fallback to load balancer policy)", invalidOverrideHost)
+
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/endpoint-override-header",
+					Headers: map[string]string{
+						"x-custom-host": invalidOverrideHost,
+					},
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			// Make multiple requests and verify fallback behavior (just check 200 response)
+			for i := 0; i < sendRequests; i++ {
+				cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+				require.NoError(t, err, "failed to get expected response")
+				require.NoError(t, http.CompareRequest(t, &req, cReq, cResp, expectedResponse), "failed to compare request and response")
+
+				// For invalid override host, we just verify the response is 200 (fallback works)
+				// No need to check specific pod routing since it should use fallback policy
+			}
+
+			t.Logf("All %d requests with invalid override host %s got 200 response (fallback working)", sendRequests, invalidOverrideHost)
+		})
+
+		t.Run("header-based endpoint override without header should fallback to load balancer policy", func(t *testing.T) {
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/endpoint-override-header",
+					// No x-custom-host header
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			// Make multiple requests and verify fallback behavior (just check 200 response)
+			for i := 0; i < sendRequests; i++ {
+				cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+				require.NoError(t, err, "failed to get expected response")
+				require.NoError(t, http.CompareRequest(t, &req, cReq, cResp, expectedResponse), "failed to compare request and response")
+
+				// For missing header, we just verify the response is 200 (fallback works)
+				// No need to check specific pod routing since it should use fallback policy
```

Comment:
should we verify that lb policy fallback to RoundRobin?

Reply:
we could do that when adding more e2e tests coverage, currently we covered if the core fallback workes

Reply:
we can track this in a separated issue.

Reply:
https://github.com/envoyproxy/gateway/issues/6634